spinalcordtoolbox/scripts/sct_maths.py

```diff
@@ -459,14 +459,16 @@ def main(argv: Sequence[str]):
                         list_data = get_data_arrays(data.shape, arg_value)
                     except ValueError as e:
                         printv(f"ERROR: -{arg_name}: {e}", 1, 'error')
-                    data += np.sum(list_data, axis=0)
+                    # the addition can't use += because the dtypes could be different
+                    data = data + np.sum(list_data, axis=0)
```

Comment:
I was curious to see if `np.sum` is susceptible, but it isn't. I guess a good rule of thumb, then, is that we should only be concerned about in-place operations (where something gets added to something else), rather than operations whether the order doesn't matter.

Reply:
It's kind of tricky! But the numpy docs do specify a lot of it. For example, `help(np.add)` shows that:
- It's possible to specify an existing array to write the output.
- It's possible to specify the casting, which defaults to `same_kind`.

For example, it's possible to trigger the casting error with `np.add`:
```py
>>> left = np.arange(3, dtype=int)
>>> right = np.arange(3, dtype=float)
>>> out = np.zeros(3, dtype=int)
>>> # regular addition, newly allocated array for the output
>>> np.add(left, right)
array([0., 2., 4.])
>>> # existing array for the output
>>> np.add(left, right, out=out)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
numpy.core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'add' output from dtype('float64') to dtype('int64') with casting rule 'same_kind'
```

Whereas `help(np.sum)` shows that:
- It's possible to specify an existing array to write the output.
- But, there's no way to specify the casting rule to use.
- However there's an implicit typecast (which can truncate values!)
  ```
    out : ndarray, optional
        Alternative output array in which to place the result. It must have
        the same shape as the expected output, but the type of the output
        values will be cast if necessary.
  ```